File: Toggly.FeatureManagement.Flutter/toggly/lib/src/services/sdk_identity.dart
import 'package:flutter/foundation.dart' show kIsWeb;

const String sdkId = 'flutter';
const String sdkVersion = '1.5.1';

String sdkUserAgent() => 'toggly-$sdkId/$sdkVersion';

Map<String, String> sdkCustomHeaders() => {
      'X-Toggly-Sdk': sdkId,
      'X-Toggly-Sdk-Version': sdkVersion,
    };

Map<String, dynamic> sdkHttpHeaders([Map<String, dynamic>? existing]) {
  final headers = <String, dynamic>{...?existing};
  if (kIsWeb) {
    headers.addAll(sdkCustomHeaders());
  } else {
    headers['User-Agent'] = sdkUserAgent();
  }
  return headers;
}

String appendSdkQueryString({String? cachedRevision}) {
  final params = <String, String>{
    'sdk': sdkId,
    'sdkVersion': sdkVersion,
  };
  if (cachedRevision != null && cachedRevision.isNotEmpty) {
    params['rev'] = cachedRevision;
  }
  return Uri(queryParameters: params).query;
}

File: Toggly.FeatureManagement.Electron/src/sdk-identity.ts
export const SDK_ID = 'electron'
export const SDK_VERSION = '1.0.0'

export const SDK_HEADER_ID = 'X-Toggly-Sdk'
export const SDK_HEADER_VERSION = 'X-Toggly-Sdk-Version'

export function sdkUserAgent(): string {
  return `toggly-${SDK_ID}/${SDK_VERSION}`
}

export function sdkCustomHeaders(): Record<string, string> {
  return {
    [SDK_HEADER_ID]: SDK_ID,
    [SDK_HEADER_VERSION]: SDK_VERSION,
  }
}

export function appendSdkQueryParams(params: URLSearchParams): void {
  params.set('sdk', SDK_ID)
  params.set('sdkVersion', SDK_VERSION)
}

export function buildDefinitionFetchHeaders(
  existing: Record<string, string> = {},
): Record<string, string> {
  return {
    ...existing,
    'User-Agent': sdkUserAgent(),
    ...sdkCustomHeaders(),
  }
}
